### PR TITLE
Improve coverage file configuration metadata

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ∅
| License       | MIT
| Doc PR        | ∅

### CODE REVIEW

File: tests/Metadata/Resource/Factory/FileConfigurationMetadataTest.php [hunk 1]
```diff
@@ -153,15 +154,11 @@ public function testXmlDoesNotExistMetadataFactory()
     /**
      * @expectedException \InvalidArgumentException
```
Comment (dunglas) [inline tests/Metadata/Resource/Factory/FileConfigurationMetadataTest.php]: We found a problem :) Should be `\ApiPlatform\Exception\InvalidArgumentException` (same for all `throw new \InvalidArgumentException`).

File: tests/Metadata/Resource/Factory/FileConfigurationMetadataTest.php [hunk 2]
```diff
@@ -238,4 +235,62 @@ public function testYamlSingleResourceMetadata(ResourceMetadata $expectedResourc
         $this->assertInstanceOf(ResourceMetadata::class, $resourceMetadata);
         $this->assertEquals($expectedResourceMetadata, $resourceMetadata);
     }
+
+    /**
+     * @expectedException \InvalidArgumentException
```
Comment (dunglas) [inline tests/Metadata/Resource/Factory/FileConfigurationMetadataTest.php]: Same here

File: tests/Metadata/Resource/Factory/FileConfigurationMetadataTest.php (hunk 1 above)
Comment (soyuka) [inline tests/Metadata/Resource/Factory/FileConfigurationMetadataTest.php]: Great :D.

File: tests/Metadata/Resource/Factory/FileConfigurationMetadataTest.php [hunk 3]
```diff
@@ -11,6 +11,7 @@
```
Comment (dunglas) [inline tests/Metadata/Resource/Factory/FileConfigurationMetadataTest.php]: Shouldn't this test be split be in 2 (XML test and YAML test, eventually sharing some code in an `abstract` test class).

Comment (soyuka) [inline tests/Metadata/Resource/Factory/FileConfigurationMetadataTest.php]: Should I split metadata / resourcenamecollection too?

I think it's fine like this but may be cleaner splitted ofc.

**Comment by soyuka** [comment]
Comments addressed :)

File: tests/Metadata/Resource/Factory/FileConfigurationMetadataTest.php (hunk 3 above)
Comment (soyuka) [inline tests/Metadata/Resource/Factory/FileConfigurationMetadataTest.php]: Done, cleaner now!

**Comment by soyuka** [comment]
100% coverage it's good to go on my end when everything is green ;)

Please tell me if you want me to squash

**Comment by dunglas** [comment]
Thanks! Can be squashed when merging.

**Comment by soyuka** [comment]
great I just added some exception messages just in case, feel free to merge ;)

**Comment by dunglas** [comment]
Thanks @soyuka

**Pull request merged**